BUNDLE_FORMAT.md

```diff
@@ -58,3 +58,92 @@ all the dependencies to the current environment.
 When sourcing this `setup.sh` it expects `COLCON_BUNDLE_PREFIX` to
 be set to the folder where the contents of `bundle.tar` have 
 been extracted.
+
+# Bundle Format - V2
+
+## Summary
+
+The output of the bundle verb is a tar file containing multiple sub-files
+
+1. version - A file with the version of the bundle file format
+1. metadata.tar - A tar archive containing various metadata
+1. N *.tar files - These tar archives contain overlay'able contents
+
+** Overlay'able - The top level of each of the archives contains a setup.sh
+that can be sources to overlay this archive's contents on top of the 
+current environment
+
+### version
+
+Contains a single version number formatted \<major version\>.\<minor version\>. The minor version
+will be incremented whenever a backwards compatible change is made. The major version will
+be incremented if there is a non-backwards compatible change to the bundle format.
+
+### metadata.tar.gz
+
+All files contained within metadata.tar are formatted as a JSON dictionary. This allows
+for backwards compatible additions to any of the files. Order is not guaranteed for this
+archive.
+
+**installers.json:**
+
+    {
+	  "apt": {
+		"installed_packages": [
+		  {
+			"version": "2.26.1",
+			"name": "pacakge1"
+		  },
+		  {
+			"version": "1.0.6-8",
+			"name": "package2"
+		  }
+		]
+	  }
+	  "pip3": {
+		"installed_packages": [
+		  {
+			"version": "1.11.13",
+			"name": "package1"
+		  },
+		  {
+			"version": "1.8.3",
+			"name": "package2"
+		  }
+		]
+	  }
+    }
+    
+**overlays.json:**
+
+	{
+		"overlays": [
+			{
+				"file_name": "bundle.tar.gz",
+				"sha256": "3c0f891fa2e98c125198afe250054609fd26e7462570ff6d3f6654625627cbf4"
+				"offset": 1000
```

Comment:
Should we define attributes like offset?

Reply:
As long as the bundle is stored in tar format, the size will be available in the header metadata defined by the tar file. It may be a good idea to include size, so that the bundle format is archive-agnostic.

Reply:
The offset is used to allow random access into a file by reading the metadata file. We should make the units of the offset clear in the documentation (and size if it's included). If the bundle happened to be a tar file, there are multiple valid interpretations of size (bytes, number of 512-byte blocks, etc).